packages/docusaurus-plugin-content-blog/src/index.js

```diff
@@ -69,30 +71,40 @@ module.exports = function(context, opts) {
           const source = path.join(blogDir, relativeSource);
           const aliasedSource = `@site/${path.relative(siteDir, source)}`;
           const blogFileName = path.basename(relativeSource);
-          // Extract, YYYY, MM, DD from the file name.
-          const filePathDateArr = blogFileName.split('-');
-          const date = new Date(
-            `${filePathDateArr[0]}-${filePathDateArr[1]}-${
-              filePathDateArr[2]
-            }T06:00:00.000Z`,
-          );
 
           const fileString = await fs.readFile(source, 'utf-8');
           const {frontMatter, excerpt} = parse(fileString);
 
+          let date;
+          // extract date and title from filename
+          const match = blogFileName.match(FILENAME_PATTERN);
+          let linkName = blogFileName.replace(/\.mdx?$/, '');
+          if (match) {
+            const [, , dateString, name] = match;
+            date = new Date(dateString);
+            linkName = name;
+          }
+          // prefer usedefined date
+          if (frontMatter.date) {
+            date = new Date(frontMatter.date);
+          }
+          // use file create time for blog
+          date = date || (await fs.fstat(source)).ctime;
```

Comment:
ctime will always change depending on when the file was changed

maybe birthtime (file creation) is better ?

Reply:
https://nodejs.org/api/fs.html#fs_stats_ctime

Reply:
yes, misunderstanding ctime, used to thought is abbr for creation time.